### PR TITLE
Replace v5.0.x docs with link to docs.open-mpi.org

### PR DESCRIPTION
**Pull request by jsquyres**
Signed-off-by: Jeff Squyres <jsquyres@cisco.com>

### CODE REVIEW

**Comment by jsquyres** [comment]
@gpaulsen Merge this PR after you merge https://github.com/open-mpi/ompi/pull/10134

**Comment by awlauria** [comment]
@gpaulsen can you review and merge

**Review by gpaulsen — APPROVED** [review]
I noticed that many of the `man1` commands do not have a corrisponding page on the website (for example `mpicc`), but they have been reorganized to a more generic page: https://docs.open-mpi.org/en/v5.0.x/man-openmpi/man1/ompi-wrapper-compiler.1.html
This should be acceptable for the website, but for the actual manpages we'll want `man mpicc` to still resolve correctly.

**Pull request merged**
